02-hello-caqti-blocking/test/repo/init_test.ml

```diff
@@ -0,0 +1,63 @@
+(*
+  Run with one of:
+
+    - PGHOST=localhost PGDATABASE=caqti_study PGPORT=5433 dune runtest --watch
+    - PGHOST=localhost PGDATABASE=caqti_study PGPORT=5433 dune build @all @runtest --watch
+ *)
+
+let%test_unit "OCaml: 1+2=3" =
+  let ( => ) = [%test_eq: Base.int] in
+  1 + 2 => 3
+
+let%test_unit "PostgreSQL: 1+2=3 (verbose, not functional)" =
+  let ( => ) = [%test_eq: Base.int] in
+  let db_add a b =
+    match Repo.Init.connect () with
+    | Error _ -> failwith "could not connect!"
+    | Ok conn -> (
+        match Repo.Exec.add conn a b with
+        | Error _ -> failwith "could not execute 'add'"
+        | Ok n -> n)
+  in
+  db_add 1 2 => 3
+
+let or_fail msg result =
+  match result with
+  | Error _ -> failwith ("could not execute: " ^ msg)
+  | Ok v -> v
+
+let%test_unit "PostgreSQL: 1+2=3 (terser, not functional)" =
+  let ( => ) = [%test_eq: Base.int] in
+  let conn = Repo.Init.connect () |> or_fail "connect" in
+  let db_add a b = Repo.Exec.add conn a b |> or_fail "add" in
+  db_add 1 2 => 3
+
+let%test_unit "PostgreSQL: 1+2=3 (verbose, functional)" =
+  let ( => ) = [%test_eq: (Base.int, Base.string) Base.Result.t] in
+  let db_add a b =
+    match Repo.Init.connect () with
+    | Error _ -> Error "could not connect!"
+    | Ok conn -> (
+        match Repo.Exec.add conn a b with
+        | Error _ -> Error "could not execute 'add'"
+        | Ok n -> Ok n)
+  in
+  db_add 1 2 => Ok 3
+
+let%test_unit "PostgreSQL: 1+2=3 (terser, functional)" =
+  let ( => ) = [%test_eq: (Base.int, Base.string) Base.Result.t] in
+  let db_add a b =
+    let ( let* ) = Result.bind in
+    let* conn = Repo.Init.connect () |> Result.map_error (fun _ -> "connect") in
+    Repo.Exec.add conn a b |> Result.map_error (fun _ -> "add")
+  in
+  db_add 1 2 => Ok 3
+
+let%test_unit "PostgreSQL: 1+2=3 (terser alternative, functional)" =
+  let ( => ) = [%test_eq: (Base.int, Base.string) Base.Result.t] in
+  let db_add a b =
+    let ( >>= ) = Result.bind in
+    Repo.Init.connect () |> Result.map_error (fun _ -> "connect")
+    >>= fun conn -> Repo.Exec.add conn a b |> Result.map_error (fun _ -> "add")
+  in
+  db_add 1 2 => Ok 3
```

Comment:
I suggest at this point, after the nice explanation of monads in the blocking section, we could adapt a preferred style, which in my opinion is the previous alternative.  I think the cases where `(>>=)` can benefit the presentation is when point-free, i.e. when the right-hand-side is not an abstraction.  There is one exception, when the abstraction is a `function` pattern-matching block, since OCaml don't have a `(match*)` operator yet.  (There has been a discussion about it in the OCaml issue tracker, but without resolution, I can find it if you are interested.)

Reply:
Thanks. The let operator syntax is indeed the preferred style from there on. Appart for the Async section where I introduce the idiomatic (for Janestreet) `>>=?` and `>>|` operators.

I think it'd be nice to leave this example here as a demonstration of the alternative syntax, just to help the curious beginner have a bit more context.

I can remove it though, if you feel it may cause more confusion than understanding.

I remember coming across `>>=?` by reading your code actually, and scratching my head for a bit :)
I had to leave it for a few weeks or months before I could come back and make sense of it.